en/docs/extended_topics/user_op.md

```diff
@@ -1,414 +1,816 @@
+//  The operator op1_name is used to calculate the gradient of myop.in
+ctx->DefineOp(op1_name, 
+  [&ctx](user_op::BackwardOpBuilder& builder) {
+    return builder.OpTypeName("multiply")
+        .InputBind("x", ctx->FwOp().input("in", 0)) //multiply.x <- myop.in
+        .InputBind("y", ctx->FwOp().output_grad("out", 0)) //multiply.y <- myop.out的梯度
+        .Output("out")
+        .Build();
+  });
+```
 
-namespace oneflow {
+Then `op2_name` is defined and use the existing operator `op2_name` to solve for `6*op1_name`.
 
-REGISTER_USER_OP_GRAD("Relu").SetGenBackwardOpConfFn([](const user_op::UserOpWrapper& op,
-                                                          user_op::AddOpFn AddOp) {
-  if (op.NeedGenGradTensor4OpInput("in", 0)) {
-    user_op::UserOpConfWrapperBuilder builder(op.op_name() + "_grad");
-    user_op::UserOpConfWrapper relu_grad_op =
-        builder.Op("relu_grad")
-            .Input("y", op.output("out", 0))
-            .Input("dy", op.GetGradTensorWithOpOutput("out", 0))
-            .Output("dx")
-            .Build();
-    op.BindGradTensorWithOpInput(relu_grad_op.output("dx", 0), "in", 0);
-    AddOp(relu_grad_op);
-  }
-});
+```cpp
+// The operator op2_name is used to calculate 6*op1_name.
+ctx->DefineOp(op2_name, 
+  [&ctx, &op1_name](user_op::BackwardOpBuilder& builder) {
+    return builder.OpTypeName("scalar_mul")
+        .InputBind("in", ctx->GetOp(op1_name).output("out", 0))
+        .Attr("has_float_operand", true)
+        .Attr("has_int_operand", false)
+        .Attr("float_operand", static_cast<double>(6))
+        .Attr("int_operand", static_cast<int64_t>(6))
+        .Output("out")
+        .Build();
+  });
+```
+
+Finally bind the output of `op2_name` (i.e., `6*x*dy`) to the input of `myop` to complete the registration.
 
-}  // namespace oneflow
+```cpp
+// (the gradient of myop.in) <- op1_name.out
+ctx->FwOp().InputGradBind(user_op::OpArg("in", 0), 
+  [&ctx, &op2_name]() -> const std::string& {
+    return ctx->GetOp(op2_name)
+          .output("out", 0);
+  });
 ```
-- 后向生成函数的步骤：
 
-宏`REGISTER_USER_OP_GRAD(op_type_name).SetGenBackwardOpConfFn(fn);`用来注册你的自定义op的后向生成函数，其中fn函数具有两个参数，`UserOpWrapper op`和`AddOpFn AddOp`，其中op表示你的自定义op，AddOp表示向整个计算图中添加一个新的op（用于后向图展开）。 在生成后向图的函数中，图里的blob是由一个叫logical blob name的字符串表示的，其中包含了产生这个blob的op的name，以及这个blob在这个op里的name。后向生成函数的任务是对前向op的输入blob，生成一个op的子图，这个子图接收前向op的输入输出blob以及输出blob的导数（梯度/grad）blob，子图的最终输出是前向op输入blob对应的导数（梯度）blob，因此针对每个（可能）需要生成梯度的blob，都需要构建一个由其他op组成的子图，并将子图的输出blob与这个需要生成梯度的blob绑定。编写这个生成子图的过程通常包含下面几步：
-  1. 判断前向op的某一个input blob是否需要生成后向的梯度blob。（我们强烈建议进行这个判断，即使你认为这个op的input一定会有梯度。因为oneflow在系统的构图优化后，可能会分析得到这个input的梯度计算是没有意义的，则可以知道不需要构建这个额外的后向子图。）
-  2. 使用UserOpConfWrapperBuilder来构建这个子图中的new_op，通常这些new_op的输入是前向op的in/out或者out对应的out_grad。
-  3. 将构建好的子图的输出blob的logical blob name与前向op的输入blob绑定
-  4. 使用AddOp函数将第2步中构建的new_op添加到计算图中
+The above is the complete process of registering a gradient and the related classes and methods will describe below.
```

Comment:
The above code is the complete process of registering a gradient and the related classes and methods will be described in below.